arbnode/dataposter/data_poster.go

```diff
@@ -236,6 +237,34 @@ func rpcClient(ctx context.Context, opts *ExternalSignerCfg) (*rpc.Client, error
 	)
 }
 
+// txToSendTxArgs converts transaction to SendTxArgs. This is needed for
+// external signer to specify From field.
+func txToSendTxArgs(addr common.Address, tx *types.Transaction) (*apitypes.SendTxArgs, error) {
+	if tx.To() == nil {
+		return nil, fmt.Errorf("transaction %v has no destination", tx.Hash())
```

Comment:
While we don't currently do this in the batch poster or validator, this is a valid type of transaction. A nil to field indicates the transaction deploys a contract.

Reply:
Done. I've also:
- refactored the code quite a bit, removed duplicate test implementation of external signer and exposed it as separate test package
- used geth server instead of doing muxing and json encoding/decoding manually.
- propagated chainID to dataposter
- added chainID to the transactions created in dataposter